wiki/DEBUGGING.md

```diff
@@ -0,0 +1,128 @@
+# An Introduction to Debugging in Haskell
+
+## Introduction
+
+There is a general purpose debugging guide available [here](https://wiki.haskell.org/Debugging), but it
+is outdated and lacks information on how to navigate a modern debugger like gdb. This guide will look two main
+cases for debugging a program in Haskell. The first being printing in the code using
+[Debug.Trace](https://hackage.haskell.org/package/base-4.21.0.0/docs/Debug-Trace.html), a built-in package that great
+for working through smaller problems. The second will focus on tracing through the code using the
+[GHCi debugger](https://downloads.haskell.org/~ghc/8.10.3/docs/html/users_guide/ghci.html#the-ghci-debugger), which
+is very closely related in function to gdb, the debugger used in C in courses like CSC209 and CSC369. **Note:** If
+there are any issues or if any new commands, tools or features end up being useful, please update this guide
+accordingly!
+
+## Debug.Trace
+
+This library is a quick and effective way to print values of objects in your Haskell Program. Please refer to the
+documentation about which specific functions to use for your use case. To be more specific, a different function will
+be called if you want to print a monad type object, a primitive type object or an algebraic datatype object. First you
+need to `import Debug.Trace` at the top of your file. The following example below shows usage for a monadic type, since
+Courseography is packed with Monads!
+
+### Example
+
+First we add `traceM` after the monad we want to view:
+
+![Image](images/debugTrace.png)
+
+Then after running the program in the terminal, we get to see our monad's value!:
+
+![Image](images/debugTraceOutput.png)
+
+## GHCi Debugger
+
+### Setup
+
+After setting Courseography as the root in terminal, type in `stack ghci`. This will set up the environment for
```

Comment:
The detail provided here is beyond the scope of this guide. You can include two things:

1. The fact that developers need to run `stack ghci` to get started (you don't need to say anything about "Courseography as the root", this is implicit in the guide).
2. Developers may need to load specific modules using `:l Module` (module names are capitalized by default).